gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/Policy.java

```diff
@@ -0,0 +1,367 @@
+  public static final class Member implements Serializable {
+
+    private static final long serialVersionUID = 6496912037577986137L;
+    private final MemberType memberType;
+    private final String emailOrDomain;
+
+    Member(MemberType memberType, String emailOrDomain) {
+      this.memberType = memberType;
+      this.emailOrDomain = emailOrDomain;
+    }
+
+    public static Member allUsers() {
+      throw new UnsupportedOperationException(
+          "Google Cloud Resource Manager does not support the \"all users\" member type yet.");
+      // return new Member(MemberType.ALL_USERS, null);
+    }
+
+    public static Member allAuthenticatedUsers() {
+      throw new UnsupportedOperationException("Google Cloud Resource Manager does not support the "
+          + "\"all authenticated users\" member type yet.");
+      // return new Member(MemberType.ALL_AUTHENTICATED_USERS, null);
+    }
+
+    public static Member user(String email) {
+      return new Member(MemberType.USER, email);
+    }
+
+    public static Member serviceAccount(String email) {
+      return new Member(MemberType.SERVICE_ACCOUNT, email);
+    }
+
+    public static Member group(String email) {
+      return new Member(MemberType.GROUP, email);
+    }
+
+    public static Member domain(String domain) {
+      throw new UnsupportedOperationException(
+          "Google Cloud Resource Manager does not support domain members yet.");
+      // return new Member(MemberType.DOMAIN, domain);
+    }
+
+    public MemberType type() {
+      return memberType;
+    }
+
+    public String emailOrDomain() {
+      return emailOrDomain;
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(memberType, emailOrDomain);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      return obj instanceof Member && Objects.equals(this.memberType, ((Member) obj).memberType)
+          && Objects.equals(this.emailOrDomain, ((Member) obj).emailOrDomain);
+    }
+
+    String toPb() {
+      return emailOrDomain != null ? memberType.prefix() + emailOrDomain : memberType.prefix();
+    }
+
+    static Member fromPb(String memberPb) {
+      String[] memberInfo = memberPb.split(":", 2);
+      String memberStr = memberInfo[0];
+      String emailOrDomain = (memberInfo.length > 1) ? memberInfo[1] : null;
+      switch (memberStr) {
+        case "allUsers":
+          return new Member(MemberType.ALL_USERS, null);
+        case "allAuthenticatedUsers":
+          return new Member(MemberType.ALL_AUTHENTICATED_USERS, null);
+        case "user":
+          return new Member(MemberType.USER, checkNotNull(emailOrDomain));
+        case "serviceAccount":
+          return new Member(MemberType.SERVICE_ACCOUNT, checkNotNull(emailOrDomain));
+        case "group":
+          return new Member(MemberType.GROUP, checkNotNull(emailOrDomain));
+        case "domain":
+          return new Member(MemberType.DOMAIN, checkNotNull(emailOrDomain));
+        default:
```

Comment:
We should support that. [e.g.](https://github.com/GoogleCloudPlatform/gcloud-java/blob/master/gcloud-java-storage/src/main/java/com/google/gcloud/storage/Acl.java#L123)